File: app/models/bill_cancellation.py
from datetime import datetime
from app.database import Column, Model, SurrogatePK, db, reference_col


class BillCancellation(Model, SurrogatePK):
    __tablename__ = "bill_cancellations"

    bill = db.relationship("Bill", back_populates="bill_cancellation")
    company_id = reference_col("companies")
    company = db.relationship("Company", foreign_keys=[company_id], back_populates="bill_cancellations")
    cancellation_time = Column(db.DateTime, nullable=False, default=datetime.utcnow)
    reason = Column(db.String, nullable=False)

File: backend/Invoicing/src/Invoicing.Domain/Entities/BillCancellation.cs
using Invoicing.Domain.Common;

namespace Invoicing.Domain.Entities;
public class BillCancellation: BaseEntity
{
    public long BillId { get; set; }
    public Bill Bill { get; set; }

    public long CompanyId { get; set; }
    public Company Company { get; set; }

    public DateTime CancellationTime { get; set; } = DateTime.UtcNow;
    public string Reason { get; set; }
}
